distribution/lib/Standard/Base/0.0.0-dev/src/Data/Enso_Cloud/Enso_File.enso

```diff
@@ -0,0 +1,196 @@
+import project.Any.Any
+import project.Data.Enso_Cloud.Utils
+import project.Data.Index_Sub_Range.Index_Sub_Range
+import project.Data.Json.JS_Object
+import project.Data.Text.Encoding.Encoding
+import project.Data.Text.Matching_Mode.Matching_Mode
+import project.Data.Text.Text
+import project.Data.Text.Text_Sub_Range.Text_Sub_Range
+import project.Data.Vector.Vector
+import project.Error.Error
+import project.Errors.File_Error.File_Error
+import project.Errors.Illegal_Argument.Illegal_Argument
+import project.Errors.Problem_Behavior.Problem_Behavior
+import project.Errors.Unimplemented.Unimplemented
+import project.Network.HTTP.HTTP
+import project.Network.HTTP.HTTP_Method.HTTP_Method
+import project.Nothing.Nothing
+import project.System.File.File_Access.File_Access
+import project.System.File_Format.File_For_Read
+import project.System.Input_Stream.Input_Stream
+import project.System.Output_Stream.Output_Stream
+from project.Data.Boolean import Boolean, False, True
+from project.Data.Text.Extensions import all
+from project.System.File_Format import Auto_Detect, File_Format, Bytes, Plain_Text_Format
+
+type Enso_File
+    ## PRIVATE
+       Represents a file or folder within the Enso cloud.
+    Value name:Text id:Text organisation:Text asset_type:Enso_Asset_Type
+
+    ## Represents the root folder of the current users.
+    root : Enso_File
+    root = Enso_File.Value "" "" "" Enso_Asset_Type.Folder
+
+    ## PRIVATE
+       Target URI for the api
+    internal_uri : Text
+    internal_uri self = case self.asset_type of
+        Enso_Asset_Type.Folder -> if self.id == "" then Utils.directory_api else (Utils.directory_api + "?parent_id=" + self.id)
+        Enso_Asset_Type.File -> if self.id == "" then Error.throw (Illegal_Argument.Error "Invalid ID for a file.") else (Utils.files_api + "?parent_id=" + self.id)
+        Enso_Asset_Type.Project -> if self.id == "" then Error.throw (Illegal_Argument.Error "Invalid ID for a project.") else (Utils.projects_api + "/" + self.id)
+        Enso_Asset_Type.Connection -> if self.id == "" then Error.throw (Illegal_Argument.Error "Invalid ID for a connection.") else (Utils.secrets_api + "/" + self.id)
+
+    ## Checks if the folder or file exists
+    exists : Boolean
+    exists self =
+        auth_header = Utils.authorization_header
+        response = HTTP.fetch self.internal_uri HTTP_Method.Get [auth_header]
+        response.status_code == 200
```

Comment:
If I read this right, we do a full `GET` request, meaning we actually start downloading the file which may be costly if the file is big + will cost us data transfer.

Can we use `HEAD` option here instead?

Reply:
In the cloud API, the files GET doesn't give the file it just gives the metadata for the file.
I have asked for a HEAD method to be added to the API to do exists check.